exp/controllers/awsmachinepool_controller.go

```diff
@@ -393,7 +393,18 @@ func (r *AWSMachinePoolReconciler) reconcileDelete(machinePoolScope *scope.Machi
 
 func (r *AWSMachinePoolReconciler) updatePool(machinePoolScope *scope.MachinePoolScope, clusterScope cloud.ClusterScoper, existingASG *expinfrav1.AutoScalingGroup) error {
 	asgSvc := r.getASGService(clusterScope)
-	if asgNeedsUpdates(machinePoolScope, existingASG) {
+
+	subnetIDs, err := asgSvc.SubnetIDs(machinePoolScope)
+	if err != nil {
+		return errors.Wrapf(err, "fail to get subnets for ASG")
+	}
+	machinePoolScope.Debug("determining if subnets change in machinePoolScope",
+		"subnets of machinePoolScope", subnetIDs,
+		"subnets of existing asg", existingASG.Subnets)
+	less := func(a, b string) bool { return a < b }
+	subnetChanges := cmp.Diff(subnetIDs, existingASG.Subnets, cmpopts.SortSlices(less)) != ""
```

Comment:
I didn't use cmp.Equal here.
Because the order of subnetIDs from `subnetIDs, err := asgSvc.SubnetIDs(machinePoolScope)` and  existingASG.Subnets are always different in some cases even if they have same elements. Then each reconcile will bring an asg update using cmp.Equal . So I use cmp.Diff to disregard the order to avoid the useless asg updates.
 
I am also good to keep cmp.Equal if experts think frequent asg updates is fine.

Reply:
I think use of cmp.Diff is fine 🙂 thanks for the explanation